### ISSUE TITLE
IndexError: index 5 is out of bounds for axis 0 with size 5

### ISSUE THREAD

**Issue by jcohenadad** [neuropoly/spinalcordtoolbox#673]
data: folder_shared/sct_issues/
commit: d215adcb82d0c86b15034c522ff9c4d33582051d
~~~
sct_process_segmentation -i gre-mask.nii.gz -p centerline
  OK: gre-mask.nii.gz

Check parameters:
.. segmentation file:             gre-mask.nii.gz

Create temporary folder...
mkdir tmp.151120142439_776173/

Copying data to tmp folder...
sct_convert -i /Users/julien/data/shim/20151120_shim_005/gre-mask.nii.gz -o tmp.151120142439_776173/segmentation.nii.gz

Orient centerline to RPI orientation...
sct_image -i segmentation.nii.gz -setorient RPI -o segmentation_RPI.nii.gz

Open segmentation volume...

Get data dimensions...
.. matrix size: 48 x 192 x 192
.. voxel size:  1.15mm x 1.14583mm x 1.14583mm
/Users/julien/miniconda/lib/python2.7/site-packages/numpy/core/_methods.py:59: RuntimeWarning: Mean of empty slice.
  warnings.warn("Mean of empty slice.", RuntimeWarning)
/Users/julien/miniconda/lib/python2.7/site-packages/numpy/core/_methods.py:71: RuntimeWarning: invalid value encountered in double_scalars
  ret = ret.dtype.type(ret / rcount)

Smooth centerline/segmentation...
.. Get center of mass of the centerline/segmentation...
.. Smoothing algo = hanning
.. Windows length = 80
WARNING: The smoothing window is larger than the number of points. New value: 5
WARNING: The smoothing window is larger than the number of points. New value: 5
Traceback (most recent call last):
  File "/Users/julien/code/spinalcordtoolbox/bin/sct_process_segmentation", line 692, in <module>
    main()
  File "/Users/julien/code/spinalcordtoolbox/bin/sct_process_segmentation", line 149, in main
    fname_output = extract_centerline(fname_segmentation, remove_temp_files, verbose=param.verbose, algo_fitting=param.algo_fitting)
  File "/Users/julien/code/spinalcordtoolbox/bin/sct_process_segmentation", line 317, in extract_centerline
    data[round(x_centerline_fit[iz-min_z_index]), round(y_centerline_fit[iz-min_z_index]), iz] = 1 # if index is out of bounds here for hanning: either the segmentation has holes or labels have been added to the file
IndexError: index 5 is out of bounds for axis 0 with size 5
~~~

N.B. I discovered the issue when running propseg

**Comment by jcohenadad** [comment]
I noticed that ``sct_propseg`` now uses ``sct_process_segmentation`` to extract centerline.
However, when specifying few points along the cord, ``sct_process_segmentation`` fails to smooth the centerline because the current implementation of **smooth_centerline** (called by ``sct_process_segmentation`` and located in ``sct_straighten_spinalcord``) removes all z that have no labels. So, if only 5 points are specified, then the matrix only smoothes with 5 points. 

This seems wrong, and I am wondering why we did not notice this problem before?  Datasets that worked in November 11th do not work now, although I don't notice specific change since that date?
